### ISSUE TITLE
[a11y-defect-2]: Date should have more specific error messaging - Found in 508 Audit on form 21-4142

### ISSUE THREAD

**Issue by tbaker1026** [department-of-veterans-affairs/va.gov-team#63437]
## Point of contact

@jeana-adhoc 

## Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

## Details

The 508 office returned an audit with this issue: When errors are automatically detected, the suggestion for fixing the error is given to the user.

1. On `Step 7 of 12: Treatment records`, in the “Last treatment date” section,  select “January” in the Month field, and enter “2013” in the Year field, leave the “Day” blank. An error message is read by assistive technology “Please enter a date”.  The error message is described to the user that the Day field needs to be corrected. (New: 08/02/2023). SME: Agree:  Just like the Year should indicate the format for the Year when entered incorrectly.  Whenever you have an error message that error message needs to describe to the user how to fix the error found or give the format for data that would be allowed. <br /><br /><img width="305" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/39f0e866-15e5-45b6-bc1a-79f0247c5029">   

## Reproduction steps
1. Navigate to Step 7 in the form, fill out the fields. Once you get to the treatment date section, fill out the month, skip the date, and fill out the year. Click "Continue". The error messages display. 

## Proposed solution or next steps
1. For the memorable date error messaging, we should be more specific in what the error is. This may mean multiple new validation messages "Please select the day" (if the day has not been selected) "Please enter four digits for the year" (if the year is not formatted correctly), "Please select a month" (if a month has not been selected)

## References, articles, or WCAG support

1. [WCAG Success criteria 3.3.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/minimize-error-identified.html)

## Type of issue

- [X] Screenreader
- [X] Keyboard
- [ ] Focus
- [ ] Headings
- [ ] Color
- [ ] Zoom
- [ ] Semantics
- [ ] Axe-core
- [X] Something else
  - [X] Error messaging

**Comment by jeana-adhoc** [comment]
I want to cross reference some work that I know the design system team has been working on around memorable date to make sure we're not duplicating efforts now.  I'll report back soon.

**Comment by jeana-adhoc** [comment]
Actually - if we could convert this to the v3 component  - I think our problems would be solved. [Looking in storybook](https://design.va.gov/storybook/?path=/docs/uswds-va-memorable-date--default), I see independent error messages for each input

Is that possible? What I saw previously was work on the v3 component - not the older one that this form is using.

**Comment by rhasselle-oddball** [comment]
It's difficult to migrate this to a web component because that would involve changing all fields on the page to a web components, and there's a lot of components on this page. Also these fields are inside an array which makes it more complicated.

I'd prefer a solution where we just update this specific scenario.

@jeana-adhoc  Can we settle on a simple solution such as just making the text 'Please enter a **complete** date' ?  

The date value is undefined if they have an incomplete date, and we intentionally chose this design based on the fact that we don't want to show an error if they tab past the field.  so when they click submit, we don't have access to knowledge of how much of the data they filled out, and it would require a larger refactor to a deprecated widget.

**Comment by jeana-adhoc** [comment]
Hi @rhasselle-oddball - This all makes sense. Thanks for looking into it and explaining the challenges so concisely.  Let's move forward with this simple solution, but can we make this the error message?

>Select a month and date. And enter a valid 4-digit year.

This should at least let users know what they need to do to make it a complete date. After this is on production, I'll let the 508 office know why we went this route.

Thanks!
Jeana

**Comment by rhasselle-oddball** [comment]
Solution notes:
❌- We did not update to use web components because there were other fields on the page, and we don't want to mix and match
❌- We didn't want to refactor the validation of the previous Widget, because it would be a little complicated and also that is deprecated
✅- The most straightforward solution we ended up doing: Update error message text to: "Select a month and day. And enter a valid 4-digit year."

**Comment by jeana-adhoc** [comment]
I've confirmed the fix on staging!

<img width="501" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/3287402c-0de5-4739-91f5-8219a8c2ec5d">

Thanks for your detailed notes here @rhasselle-oddball, they're really helpful as I write back to the 508 office!
